### ISSUE TITLE
Camera transformation

### ISSUE THREAD

**Issue by eliabntt** [introlab/rtabmap#488]
Hi, I have peeked into the source code but I can't find what I'm looking for. 
I'm currently doing visual rgbd odometry with your library.

As long as I understand rtabmap in the rgbd case subscribe to the cameraInfo message to get the necessary parameters to transform the inputs (rgb/depth or rgbd sync) into cloud, obstacles etc.
Now I have tried to fake this cameraInfo message applying a fake dynamic rotation to the P matrix(camera matrix) with the usual convention P = K[R|t] but it doesn't work.
How can I tell to your system to project the current cloud for example on the right of the robot?

In short I'd like to rotate my camera without rotating the robot, so robot is going straight camera is tilting. To project the cloud I suppose you use the direction given by the frame_id parameter so if this rotate I may have chances that the cloud rotate and the visual odometry is still good? or may I pass something to the cameraInfo topic to do so? I can also change some functions in your code if you can point them out.

Cheers,
Elia

**Comment by matlabbe** [comment]
You can just modify TF frame between base and the camera. As a general example, we could have a camera on the head of a humanoid robot and visual odometry computed with that camera. As long as TF between the torso and the head is updated when rotating/tilting the head, visual odometry will adjust the transform to base frame. Visual odometry `frame_id` should be the base frame of the robot, not the camera to work. Example: if the robot is not moving, but the head is rotating, visual odometry will still say that the robot is not moving because it knows that the same rotation seen in the camera is the inverse of the TF between torso and head.

cheers,
Mathieu
